### ISSUE TITLE
tabix index loading issue

### ISSUE THREAD

**Issue by fanwayne** [samtools/htslib#1033]
#### Are you using the latest version of [samtools](https://github.com/samtools/samtools/releases/latest) and [HTSlib](https://github.com/samtools/htslib/releases/latest)? If not, please specify.
(run `samtools --version`)
samtools 1.10
Using htslib 1.10.2
Copyright (C) 2019 Genome Research Ltd.

 #### Please describe your environment.
- OS (run `uname -sr` on Linux/Mac OS or ` wmic os get Caption, Version` on Windows)
Darwin 19.3.0
- machine architecture (run `uname -m` on Linux/Mac OS or ` wmic os get OSArchitecture` on Windows)
x86_64
- compiler (run `gcc --version` or `clang --version`)
Apple clang version 11.0.0 (clang-1100.0.33.16)
Target: x86_64-apple-darwin19.3.0
Thread model: posix

#### Please specify the steps taken to generate the issue, the command you are running and the relevant output.
I was trying to retrieve the rs identifier from the genome data on NCBI (ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF) with the following command.

`awk '{gsub("chr","",$1)} {print $1":"$2"-"$3}' coord.txt | parallel --delay 1 tabix ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz {} | awk -v OFS="\t" 'BEGIN {print "chromosome","Coordinate","rsID","Ref","Alt"} {print "chr"$1,$2,$3,$4,$5}'
`

However, the tabix command did not work well and generated the following log issues. 

> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
> [E::hts_idx_load3] Could not load local index file '00-All.vcf.gz.tbi'
> Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
 **[Above Repeated....]**

Here is a small sample data of the "coord.txt" file.

> 1	17375	17375
> 1	17407	17407
> 1	564542	564542
> 1	564621	564621
> 1	564667	564667
> 1	565184	565184
> 1	565292	565292
> 1	565400	565400
> 1	565433	565433
> 1	565697	565697

I want to know how it happens and how to fix it. Thank you very much for your help.

**Comment by daviesrob** [comment]
Issue transferred to htslib, as tabix is part of that project.

Do you have write access to the directory you were in when you ran tabix?  It will try to download the index file `00-All.vcf.gz.tbi` there, and may complain if it can't do that.

It's also possible that more than one of your parallel tabix runs tried to write to the same local file and corrupted it.  If you do have a local `00-All.vcf.gz.tbi` file, try removing it and getting it again by running a single `tabix` command, say,

```
tabix ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz 1:17375-17375
```

If it works you should have a good copy of the index and your parallel run will be able to use it.

You may also want to increase the delay in your parallel command, in case the NCBI's FTP site is having trouble keeping up.

**Comment by fanwayne** [comment]
Hi @daviesrob, many thanks for your support and suggestion. I confirm I have the access to the working directory. There is no local 00-All.vcf.gz.tbi in the directory. I aslo checked the single tabix command, but it did not work.

**Comment by valeriuo** [comment]
Did the single `tabix` command fail with the same error?
What happens when you run the command `touch 00-All.vcf.gz` in the same folder?

**Comment by fanwayne** [comment]
Hi @valeriuo,
Running the tabix command will generate the following error:
[E::idx_test_and_fetch] Error reading "ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi"
tabix(77080,0x104939dc0) malloc: *** error for object 0x104974000: pointer being freed was not allocated
tabix(77080,0x104939dc0) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      tabix  1:17375-17375


The touch command output nothing on terminal.

**Comment by valeriuo** [comment]
Can you show the output of the following two commands?
`tabix --version`
`which tabix`

**Comment by fanwayne** [comment]
Hi, @valeriuo . Below are the outputs, thank you.
**tabix (htslib) 1.10.2
Copyright (C) 2019 Genome Research Ltd.
/usr/local/bin/tabix**

**Comment by valeriuo** [comment]
@fanwayne 
We made a fix for the crash that you reported, but the underlying cause (the failure to read **ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi**) is still present.

**Comment by valeriuo** [comment]
Can you manually download the TBI file and rerun the single region tabix command?

`wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi`

`tabix ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz 1:17375-17375`

**Comment by fanwayne** [comment]
Hi @valeriuo , this solution works!!! Thank you so much for you help!! It outputs the following text on terminal. 
**1	17375	rs755771866	A	G	.	.	RS=755771866;RSPOS=17375;dbSNPBuildID=144;SSR=0;SAO=0;VP=0x050100080005000002000100;GENEINFO=WASH7P:653635|MIR6859-1:102466751;WGT=1;VC=SNV;SLO;INT;ASP;TOPMED=0.99468813710499490,0.00531186289500509**

**Comment by valeriuo** [comment]
Can you also try running the fixed tabix without the downloaded index file?

```
rm 00-All.vcf.gz.tbi
git checkout -b tabix-fix develop
git pull https://github.com/valeriuo/htslib.git index-1033
make clean; make
./tabix ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz 1:17375-17375
```

**Comment by fanwayne** [comment]
Do I need to clone the entire repository?
When I runs it directly, it outputs the following errors.

fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
make: *** No rule to make target `clean'.  Stop.
make: *** No targets specified and no makefile found.  Stop.
zsh: no such file or directory: ./tabix

**Comment by valeriuo** [comment]
> Do I need to clone the entire repository?

My mistake. I was under the assumption you already did that. 
In that case you need to clone the repo in a separate folder. Something like:
```
git clone --single-branch --branch index-1033 https://github.com/valeriuo/htslib.git github-htslib
cd github-htslib
autoheader
autoconf
./configure
make
./tabix ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz 1:17375-17375
```

**Comment by fanwayne** [comment]
Below are the log outputs. Seems not working.

Cloning into 'github-htslib'...
remote: Enumerating objects: 83, done.
remote: Counting objects: 100% (83/83), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 12743 (delta 25), reused 59 (delta 15), pack-reused 12660
Receiving objects: 100% (12743/12743), 9.43 MiB | 604.00 KiB/s, done.
Resolving deltas: 100% (9135/9135), done.
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for ranlib... ranlib
checking for grep that handles long lines and -e... /usr/bin/grep
checking for C compiler warning flags... -Wall
checking for pkg-config... /usr/local/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking shared library type for unknown-Darwin... Darwin dylib
checking whether the compiler accepts -fvisibility=hidden... yes
checking how to run the C preprocessor... gcc -E
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/param.h... yes
checking for getpagesize... yes
checking for working mmap... yes
checking for gmtime_r... yes
checking for fsync... yes
checking for drand48... yes
checking for srand48_deterministic... no
checking whether fdatasync is declared... no
checking for library containing log... none required
checking for zlib.h... yes
checking for inflate in -lz... yes
checking for library containing recv... none required
checking for bzlib.h... yes
checking for BZ2_bzBuffToBuffCompress in -lbz2... yes
checking for lzma.h... yes
checking for lzma_easy_buffer_encode in -llzma... yes
checking for libdeflate.h... no
checking for libdeflate_deflate_compress in -ldeflate... no
checking for curl_easy_pause in -lcurl... yes
checking for CCHmac... yes
checking whether PTHREAD_MUTEX_RECURSIVE is declared... yes
configure: creating ./config.status
config.status: creating config.mk
config.status: creating htslib.pc.tmp
config.status: creating config.h
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o kfunc.o kfunc.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o knetfile.o knetfile.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o kstring.o kstring.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o bgzf.o bgzf.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o errmod.o errmod.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o faidx.o faidx.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o header.o header.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hfile.o hfile.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hfile_net.o hfile_net.c
echo '#define HTS_VERSION_TEXT "1.10.2-45-g34d8c2a"' > version.h
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hts.o hts.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hts_os.o hts_os.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o md5.o md5.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o multipart.o multipart.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o probaln.o probaln.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o realn.o realn.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o regidx.o regidx.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o region.o region.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o sam.o sam.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o vcf_sweep.o vcf_sweep.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o tbx.o tbx.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o textutils.o textutils.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o thread_pool.o thread_pool.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o vcf.o vcf.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o vcfutils.o vcfutils.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_decode.o cram/cram_decode.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_encode.o cram/cram_encode.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_external.o cram/cram_external.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_index.o cram/cram_index.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_io.o cram/cram_io.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/cram_stats.o cram/cram_stats.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/mFILE.o cram/mFILE.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/rANS_static.o cram/rANS_static.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o cram/string_alloc.o cram/string_alloc.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hfile_libcurl.o hfile_libcurl.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hfile_gcs.o hfile_gcs.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hfile_s3.o hfile_s3.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o hfile_s3_write.o hfile_s3_write.c
ar -rc libhts.a kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o header.o hfile.o hfile_net.o hts.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o region.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/string_alloc.o   hfile_libcurl.o hfile_gcs.o hfile_s3.o hfile_s3_write.o
ranlib libhts.a
gcc -dynamiclib -install_name /usr/local/lib/libhts.3.dylib -current_version 1.10.255 -compatibility_version 3 -fvisibility=hidden  -o libhts.dylib kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o header.o hfile.o hfile_net.o hts.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o region.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/string_alloc.o   hfile_libcurl.o hfile_gcs.o hfile_s3.o hfile_s3_write.o -llzma -lbz2 -lz   -lcurl 
ln -sf libhts.dylib libhts.3.dylib
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o bgzip.o bgzip.c
gcc -fvisibility=hidden  -o bgzip bgzip.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o htsfile.o htsfile.c
gcc -fvisibility=hidden  -o htsfile htsfile.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o tabix.o tabix.c
gcc -fvisibility=hidden  -o tabix tabix.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/hts_endian.o test/hts_endian.c
gcc -fvisibility=hidden  -o test/hts_endian test/hts_endian.o -llzma -lbz2 -lz   -lcurl 
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/fieldarith.o test/fieldarith.c
gcc -fvisibility=hidden  -o test/fieldarith test/fieldarith.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/hfile.o test/hfile.c
gcc -fvisibility=hidden  -o test/hfile test/hfile.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/pileup.o test/pileup.c
gcc -fvisibility=hidden  -o test/pileup test/pileup.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/sam.o test/sam.c
gcc -fvisibility=hidden  -o test/sam test/sam.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test_bgzf.o test/test_bgzf.c
gcc -fvisibility=hidden  -o test/test_bgzf test/test_bgzf.o libhts.a -lz -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test_kstring.o test/test_kstring.c
gcc -fvisibility=hidden  -o test/test_kstring test/test_kstring.o libhts.a -lz -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test_realn.o test/test_realn.c
gcc -fvisibility=hidden  -o test/test_realn test/test_realn.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test-regidx.o test/test-regidx.c
gcc -fvisibility=hidden  -o test/test-regidx test/test-regidx.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test_str2int.o test/test_str2int.c
gcc -fvisibility=hidden  -o test/test_str2int test/test_str2int.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test_view.o test/test_view.c
gcc -fvisibility=hidden  -o test/test_view test/test_view.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test_index.o test/test_index.c
gcc -fvisibility=hidden  -o test/test_index test/test_index.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test-vcf-api.o test/test-vcf-api.c
gcc -fvisibility=hidden  -o test/test-vcf-api test/test-vcf-api.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test-vcf-sweep.o test/test-vcf-sweep.c
gcc -fvisibility=hidden  -o test/test-vcf-sweep test/test-vcf-sweep.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test-bcf-sr.o test/test-bcf-sr.c
gcc -fvisibility=hidden  -o test/test-bcf-sr test/test-bcf-sr.o libhts.a -lz -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/fuzz/hts_open_fuzzer.o test/fuzz/hts_open_fuzzer.c
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test-bcf-translate.o test/test-bcf-translate.c
gcc -fvisibility=hidden  -o test/test-bcf-translate test/test-bcf-translate.o libhts.a -lz -llzma -lbz2 -lz   -lcurl  -lpthread
gcc -Wall -g -O2 -fvisibility=hidden -I.  -c -o test/test-parse-reg.o test/test-parse-reg.c
gcc -fvisibility=hidden  -o test/test-parse-reg test/test-parse-reg.o libhts.a -llzma -lbz2 -lz   -lcurl  -lpthread
sed -n '/^static_libs=/s/[^=]*=/HTSLIB_static_LIBS = /p;/^static_ldflags=/s/[^=]*=/HTSLIB_static_LDFLAGS = /p' htslib.pc.tmp > htslib_static.mk
sed -e 's#@-includedir@#'`pwd`'#g;s#@-libdir@#'`pwd`'#g' htslib.pc.tmp > htslib-uninstalled.pc
[E::idx_test_and_fetch] Error reading "ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi"
[E::idx_find_and_load] Could not retrieve index file for 'ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz'
Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz: Broken pipe

**Comment by valeriuo** [comment]
What version of libcurl is installed on your system? 
You can run `ls /usr/lib/libcurl*` to find out.

**Comment by fanwayne** [comment]
I'm not sure which version I am currently using. Below are the outputs from  "ls /usr/lib/libcurl*" .

/usr/lib/libcurl.3.dylib	/usr/lib/libcurl.dylib
/usr/lib/libcurl.4.dylib

**Comment by valeriuo** [comment]
Please run `tabix` with `--verbosity` argument set to 9:

`./tabix --verbosity 9 ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz 1:17375-17375`

This should offer more information about what libcurl is doing in the background.

**Comment by fanwayne** [comment]
Hi @valeriuo . Below are the outputs. It didn't show which version of libcurl I was using.
By the way, it is really slow for me to download the index file from ftp.ncbi.nih.gov. I'm not sure whether these issues are related. However, after the index file is downloaded, the tabix command works well.

```

[D::init_add_plugin] Loaded "knetfile"
[D::init_add_plugin] Loaded "mem"
[D::init_add_plugin] Loaded "libcurl"
[D::init_add_plugin] Loaded "gcs"
[D::init_add_plugin] Loaded "s3"
[D::init_add_plugin] Loaded "s3w"
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
*   Trying 130.14.250.12...
* TCP_NODELAY set
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
< 220-
<  This warning banner provides privacy and security notices consistent with 
<  applicable federal laws, directives, and other federal guidance for accessing 
<  this Government system, which includes all devices/storage media attached to 
<  this system. This system is provided for Government-authorized use only. 
<  Unauthorized or improper use of this system is prohibited and may result in 
<  disciplinary action and/or civil and criminal penalties. At any time, and for 
<  any lawful Government purpose, the government may monitor, record, and audit 
<  your system usage and/or intercept, search and seize any communication or data 
<  transiting or stored on this system. Therefore, you have no reasonable 
<  expectation of privacy. Any communication or data transiting or stored on this 
<  system may be disclosed or used for any lawful Government purpose.
< 220 FTP Server ready.
> USER anonymous
< 331 Anonymous login ok, send your complete email address as your password
> PASS ftp@example.com
< 230 Anonymous access granted, restrictions apply
> PWD
< 257 "/" is the current directory
* Entry path is '/'
> EPSV
* Connect data stream passively
* ftp_perform ends with SECONDARY: 0
< 229 Entering Extended Passive Mode (|||50350|)
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
* Connecting to 2607:f220:41e:250::11 (2607:f220:41e:250::11) port 50350
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
> TYPE I
< 200 Type set to I
> SIZE snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
< 213 15718889328
> RETR snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
< 150 Opening BINARY mode data connection for snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz (15718889328 bytes)
* Maxdownload = -1
* Getting file with size: 15718889328
[I::hts_idx_check_local] Using alignment file '00-All.vcf.gz'
* Hostname ftp.ncbi.nih.gov was found in DNS cache
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
*   Trying 130.14.250.12...
* TCP_NODELAY set
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
< 220-
<  This warning banner provides privacy and security notices consistent with 
<  applicable federal laws, directives, and other federal guidance for accessing 
<  this Government system, which includes all devices/storage media attached to 
<  this system. This system is provided for Government-authorized use only. 
<  Unauthorized or improper use of this system is prohibited and may result in 
<  disciplinary action and/or civil and criminal penalties. At any time, and for 
<  any lawful Government purpose, the government may monitor, record, and audit 
<  your system usage and/or intercept, search and seize any communication or data 
<  transiting or stored on this system. Therefore, you have no reasonable 
<  expectation of privacy. Any communication or data transiting or stored on this 
<  system may be disclosed or used for any lawful Government purpose.
< 220 FTP Server ready.
> USER anonymous
< 331 Anonymous login ok, send your complete email address as your password
> PASS ftp@example.com
< 230 Anonymous access granted, restrictions apply
> PWD
< 257 "/" is the current directory
* Entry path is '/'
> EPSV
* Connect data stream passively
* ftp_perform ends with SECONDARY: 0
< 229 Entering Extended Passive Mode (|||50011|)
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
* Connecting to 2607:f220:41e:250::11 (2607:f220:41e:250::11) port 50011
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
> TYPE I
< 200 Type set to I
> SIZE snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.csi
< 550 snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.csi: No such file or directory
> RETR snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.csi
< 550 snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.csi: No such file or directory
* RETR response: 550
* Remembering we are in dir ""
* Connection #0 to host ftp.ncbi.nih.gov left intact
> QUIT
< 221 Goodbye.
* Closing connection 0
[I::idx_test_and_fetch] Failed to open index file 'ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.csi'
* Hostname ftp.ncbi.nih.gov was found in DNS cache
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
*   Trying 130.14.250.12...
* TCP_NODELAY set
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
< 220-
<  This warning banner provides privacy and security notices consistent with 
<  applicable federal laws, directives, and other federal guidance for accessing 
<  this Government system, which includes all devices/storage media attached to 
<  this system. This system is provided for Government-authorized use only. 
<  Unauthorized or improper use of this system is prohibited and may result in 
<  disciplinary action and/or civil and criminal penalties. At any time, and for 
<  any lawful Government purpose, the government may monitor, record, and audit 
<  your system usage and/or intercept, search and seize any communication or data 
<  transiting or stored on this system. Therefore, you have no reasonable 
<  expectation of privacy. Any communication or data transiting or stored on this 
<  system may be disclosed or used for any lawful Government purpose.
< 220 FTP Server ready.
> USER anonymous
< 331 Anonymous login ok, send your complete email address as your password
> PASS ftp@example.com
< 230 Anonymous access granted, restrictions apply
> PWD
< 257 "/" is the current directory
* Entry path is '/'
> EPSV
* Connect data stream passively
* ftp_perform ends with SECONDARY: 0
< 229 Entering Extended Passive Mode (|||50274|)
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
* Connecting to 2607:f220:41e:250::11 (2607:f220:41e:250::11) port 50274
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
> TYPE I
< 200 Type set to I
> SIZE snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.csi
< 550 snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.csi: No such file or directory
> RETR snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.csi
< 550 snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.csi: No such file or directory
* RETR response: 550
* Remembering we are in dir ""
* Connection #0 to host ftp.ncbi.nih.gov left intact
> QUIT
< 221 Goodbye.
* Closing connection 0
[I::idx_test_and_fetch] Failed to open index file 'ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.csi'
* Hostname ftp.ncbi.nih.gov was found in DNS cache
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
*   Trying 130.14.250.12...
* TCP_NODELAY set
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
< 220-
<  This warning banner provides privacy and security notices consistent with 
<  applicable federal laws, directives, and other federal guidance for accessing 
<  this Government system, which includes all devices/storage media attached to 
<  this system. This system is provided for Government-authorized use only. 
<  Unauthorized or improper use of this system is prohibited and may result in 
<  disciplinary action and/or civil and criminal penalties. At any time, and for 
<  any lawful Government purpose, the government may monitor, record, and audit 
<  your system usage and/or intercept, search and seize any communication or data 
<  transiting or stored on this system. Therefore, you have no reasonable 
<  expectation of privacy. Any communication or data transiting or stored on this 
<  system may be disclosed or used for any lawful Government purpose.
< 220 FTP Server ready.
> USER anonymous
< 331 Anonymous login ok, send your complete email address as your password
> PASS ftp@example.com
< 230 Anonymous access granted, restrictions apply
> PWD
< 257 "/" is the current directory
* Entry path is '/'
> EPSV
* Connect data stream passively
* ftp_perform ends with SECONDARY: 0
< 229 Entering Extended Passive Mode (|||50019|)
*   Trying 2607:f220:41e:250::11...
* TCP_NODELAY set
* Connecting to 2607:f220:41e:250::11 (2607:f220:41e:250::11) port 50019
* Connected to ftp.ncbi.nih.gov (2607:f220:41e:250::11) port 21 (#0)
> TYPE I
< 200 Type set to I
> SIZE snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi
< 213 2823614
> RETR snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi
< 150 Opening BINARY mode data connection for snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi (2823614 bytes)
* Maxdownload = -1
* Getting file with size: 2823614
[I::idx_test_and_fetch] Downloading file ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi to local directory
* transfer closed with 2594134 bytes remaining to read
* Closing connection 0
[E::idx_test_and_fetch] Error reading "ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi"
[E::idx_find_and_load] Could not retrieve index file for 'ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz'
Could not load .tbi/.csi index of ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz: Broken pipe
```

**Comment by daviesrob** [comment]
> * Getting file with size: 2823614
> [I::idx_test_and_fetch] Downloading file ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz.tbi to local directory
> * transfer closed with 2594134 bytes remaining to read

It looks like the NCBI (or something between you and the NCBI) hung up before you got the whole file.

The NCBI serve this data via `https` as well as `ftp`, so you could try 
```
https://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz
```
instead to see if it's more reliable.  Otherwise, it would be best to download the files with something like `wget` which will retry failed downloads until it either gets the whole file or hits its retry limit.

**Comment by fanwayne** [comment]
Hi @daviesrob . Changing to `https` works! Thank you so much for your support.
